File: py/bit_counting.py
"""
Write a function that takes an integer as input, and returns the number of bits that are equal to one in the binary representation of that number. You can guarantee that input is non-negative.

Example: The binary representation of 1234 is 10011010010, so the function should return 5 in this case

https://www.codewars.com/kata/526571aae218b8ee490006f4/python
"""

def count_bits(n):
    # Python's string format method can take a format spec. 
    s = "{0:b}".format(n)
    count = 0 
    for char in s:
        if char == "1":
            count += 1 

    return count 
    
def log(actual, expected):
    if actual == expected:
        print("PASS {} ----------> {} ".format( actual, expected ))
    else:
        print("PASS {} ----------> {} ".format( actual, expected ))

log(count_bits(0), 0)
log(count_bits(4), 1)
log(count_bits(7), 3)
log(count_bits(9), 2)
log(count_bits(10), 2)

File: js/bit_counting.js

// https://www.codewars.com/kata/578553c3a1b8d5c40300037c/javascript
// https://www.codewars.com/kata/526571aae218b8ee490006f4/javascript
// Given an array of ones and zeroes, convert the equivalent binary value to an integer.

const count_bits = n => n.toString(2)
                        .split('')
                        .filter(ele => ele == 1)
                        .length


const MINE_countBits = (n) => {
    let count = 0
    const s = (n >>> 0).toString(2)
    arr = s.split("")
    arr.forEach((char)=>{
        if ( char == "1") {
            count++
        }
    })
    return count
}

const log = (actual, expect) => { 
    if (actual == expect) {
        console.log(`PASS ${actual} --------------> ${expect} `)
    }
    else {
        console.log(`FAIL ${actual} --------------> ${expect} `)
    }
}

log(count_bits(0), 0)
log(count_bits(4), 1)
log(count_bits(7), 3)
log(count_bits(9), 2)
log(count_bits(10), 2)
